File: concept/17_recursion/binary_search/is_sorted/is_sorted.dart
bool isSorted(List<int> arr, int length) {
  print("ss $arr");

  if (length == 0 || length == 1) {
    return true;
  }

  if (arr[0] > arr[1]) {
    return false;
  } else {
    bool ans = isSorted(arr.skip(1).toList(), length - 1);
    print(arr);
    return ans;
  }
}

void main() {
  List<int> arr = [1, 3, 4, 5, 8]; // Sorted
  List<int> arr2 = [1, 3, 4, 50, 8]; // Not Sorted

  bool isSort = isSorted(arr, 5);
  bool isSort2 = isSorted(arr2, 5);
  if (isSort) {
    print("Sorted");
  } else {
    print("Not Sorted");
  }
  if (isSort2) {
    print("Sorted");
  } else {
    print("Not Sorted");
  }
}

File: concept/17_recursion/binary_search/is_sorted/is_sorted.cpp
#include <iostream>
using namespace std;

bool isSorted(int arr[], int length)
{
    if (length == 0 || length == 1)
    {
        return true;
    }

    if (arr[0] > arr[1])
    {
        return false;
    }
    else
    {
        bool ans = isSorted(arr + 1, length - 1);
        return ans;
    }
}

int main()
{
    int arr[5] = {1, 3, 4, 5, 8};   // Sorted
    int arr2[5] = {1, 3, 4, 50, 8}; // Not Sorted

    bool isSort = isSorted(arr, 5);
    bool isSort2 = isSorted(arr2, 5);
    if (isSort)
    {
        cout << "Sorted" << endl;
    }
    else
    {
        cout << "Not Sorted" << endl;
    }
    if (isSort2)
    {
        cout << "Sorted" << endl;
    }
    else
    {
        cout << "Not Sorted" << endl;
    }
}
